src/components/Canvas/DataLayer.tsx

```diff
@@ -29,6 +30,8 @@ export default class DataLayer extends BaseLayer {
   private layers: Array<DottingDataLayer>;
   private currentLayer: DottingDataLayer;
   private defaultPixelColor = DefaultPixelColor;
+  private capturedImageBitmap: ImageBitmap | null = null;
+  private capturedImageBitmapScale = 1;
```

Comment:
From what I know, you cannot use image related APIs in an offscreen canvas. Actually I am not an expert of Offscreen canvas or workers, so I am not sure if how I implemented this feature is the correct way to do it. `ImageBitmap` is created in the offscreen canvas, and from what I know logic in offscreen canvas happens in the web worker. 

Maybe we have to add worker logic to this feature like the example below?
https://tech.kakao.com/2021/06/02/frontend-growth-09/

What is your opinion?

Reply:
For me it would be great to separate files and interest(what should have in canvas itself/ what should happen in worker API) and I assume creating offscreencanvas in the workerAPI and controlling sequence there(with postMessage & onMessage method) will optimize the overall flow.
For instance `lastRenderAllCall` can be replaced with onMessage, by giving permission of the offscreen logic to the workerAPI

Reply:
Great I will change the code as you have suggested